### PR TITLE
tart 0.33.0 (new formula)

### PR DESCRIPTION
**Pull request by umireon**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/tart.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    url "https://github.com/cirruslabs/softnet/archive/refs/tags/0.3.0.tar.gz"
+    sha256 "b77fb9424cf6a5c61e2a2513a531f0a9321047890979ca7282a5c036ebf48938"
+  end
+
+  def install
+    resource("softnet").stage do
+      system "cargo", "install", *std_cargo_args
+    end
+    system "swift", "build", "--disable-sandbox", "-c", "release"
+    system "/usr/bin/codesign", "-f", "-s", "-", "--entitlement", "Resources/tart.entitlements", ".build/release/tart"
```
Comment (SMillerDev) [inline Formula/tart.rb:26]: This should really be fixed upstream instead of having Homebrew sign files

File: Formula/tart.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    bin.install ".build/release/tart"
+  end
+
+  test do
+    ENV["TART_HOME"] = testpath/".tart"
+    empty_ipsw_path = testpath/"empty.ipsw"
+    touch empty_ipsw_path
+    output = shell_output("tart create --from-ipsw #{empty_ipsw_path} test || true")
```
Comment (SMillerDev) [inline Formula/tart.rb:34]:
```suggestion
    (testpath/"empty.ipsw").write ""
    output = shell_output("tart create --from-ipsw #{testpath/"empty.ipsw"} test", 1)
```

File: Formula/tart.rb (hunk 1 above)
Comment (umireon) [reply Formula/tart.rb:26]: If adding the following Makefile to upstream, will this formula be accepted?

```makefile
PREFIX ?= /usr/local

.PHONY: all build install

all: build

build:
	swift build --disable-sandbox -c release
	codesign -f -s - --entitlement Resources/tart.entitlements .build/release/tart

install: all
	mkdir -p "$(PREFIX)/bin"
	cp -f .build/release/tart "$(PREFIX)/bin/tart"
```

**Comment by umireon** [comment]
@SMillerDev Adding the following file to upstream and calling `make install PREFIX=#{prefix}` is acceptable?

```makefile
PREFIX ?= /usr/local

.PHONY: all build install

all: build

build:
        swift build --disable-sandbox -c release
        codesign -f -s - --entitlement Resources/tart.entitlements .build/release/tart

install: build
        install -d "$(PREFIX)/bin"
        install -m 0755 .build/release/tart "$(PREFIX)/bin/tart"
```

**Review by carlocab — DISMISSED** [review]
Ad-hoc signatures seem ok by me.

**Review by SMillerDev — DISMISSED** [review]

**Comment by umireon** [comment]
@carlocab @SMillerDev Squashed and rebased!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**